File: CarlaErik/UvA.SoftCon.Questionnaire/QL/AST/Model/Statements/Statement.cs
using System.Collections.Generic;
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QL.AST.Model.Statements
{
    public abstract class Statement : QLNode
    {
        protected Statement(TextPosition position)
            : base(position) { }

        internal abstract void CollectQuestions(ICollection<Question> questions);
    }
}

File: encoders/ql/src/main/java/org/uva/sea/ql/encoders/ast/statement/Statement.java
package org.uva.sea.ql.encoders.ast.statement;

import java.util.Collection;

import org.uva.sea.ql.encoders.ast.AstNode;
import org.uva.sea.ql.encoders.ast.TextLocation;

public abstract class Statement extends AstNode {

	public Statement(TextLocation textLocation) {
		super(textLocation);
	}

	public abstract void collectQuestions(Collection<Question> questions);

}
